### DISCUSSION TITLE
HomeKit integration

### DISCUSSION THREAD

**Discussion by noelhibbard** [mikebrady/shairport-sync#1824] [General]
I'm sure Mike will hate this topic, hahaha. So every few iOS releases I test HomeKit integration again with shairport-sync where I remove all the speakers from HomeKit and then re-add them. Well this morning I just did this and I was able to use Siri to "play in the Living Room" and "Play in the Living Room and Bathroom" or "Set the Living Room volume to 50%" or "lower the volume in the living room" or "add the bathroom". All these things were working perfectly for about 10mins and then every single speaker went to "not responding". If I remove them all the then re-add them all, it will work again for about 10mins.

I feel like this is so close to working, I feel like a little more attention and it would be reliable.

If I recall, Mike wasn't really the one to add the HomeKit integration, maybe someone else can chime in here.

This would be so nice to get working because it would even set the volume on my AVR via shairport-sync shell commands. Previously I had to hack up volume control on my AVR by adding it as a dimmable light with homebridge and the voice commands had to be really specific because it didn't think of it as a speaker but as a light.

**Comment by mikebrady** [comment]
You're right, @noelhibbard 😊. It seems as if changes are afoot in AirPlay 2. Still trying to figure some of it out.😏

**Comment by MaKin211** [comment]
> This would be so nice to get working because it would even set the volume on my AVR via shairport-sync shell commands. Previously I had to hack up volume control on my AVR by adding it as a dimmable light with homebridge and the voice commands had to be really specific because it didn't think of it as a speaker but as a light.

Would you mind sharing how you control your AVR's volume (master volume) via your iOS device? Did you configure it with HDMI CEC? I reckon that with the shairport-sync events you can only react to changes but cannot really tell whether it's a volume up or volume down event and with CEC you could not set a distinct volume level afaik. Would really appreciate your input on this.

**Reply by noelhibbard** [reply]
```run_this_when_volume_is_set``` is called on volume changes and it includes the volume level. I guess if you need a vol up and vol down events independently, you could use MQTT to monitor the events. This all assumes your AVR has some sort of IP control.

What AVR do you have? I've controlled Pioneer/Denon and Sony AVRs from shairport-sync with no issues. In addition to volume, I also automate the power and input so all I do is AirPlay and the AVR turns on and goes to the correct input.
